compiler/resolution/functionResolution.cpp

```diff
@@ -7601,6 +7639,17 @@ static void instantiate_default_constructor(FnSymbol* fn) {
         }
       }
     }
+
+    if (AggregateType* ct = toAggregateType(fn->_this->type)) {
+      if (ct->instantiationStyle == DEFINES_INITIALIZER &&
```

Comment:
I don't think I understand why this code would ever run... I thought we wouldn't create a default constructor if instantiation style is DEFINES_INITIALIZER, because the only way it would be DEFINES_INITIALIZER is if it had a user-provided initializer. Plus, it's in "instantiate_default_constructor" which makes me think it applies to generic types (which you don't support yet?)